openwisp_monitoring/monitoring/base/models.py

```diff
@@ -83,6 +83,7 @@ def _get_or_create(cls, **kwargs):
             metric = cls.objects.get(**kwargs)
             created = False
         except cls.DoesNotExist:
+            kwargs['is_healthy'] = True
```

Comment:
This field is NULL by default which means unknown.
Setting it to True is flagging the metric as healthy but it's hiding an underlying issue: it is not a good solution. The broken code is still lying around.

Why are recovery alerts being sent when the metric is created?

That shouldn't happen and I'm not sure if it's an underlying issue that was present before or if it's new, have you found out? Is this happening also with ping metrics?

Please look for a general solution, try also with ping metrics.

Reply:
I think it was present from the beginning. It should be solved by the latest iteration (which should be full-proof hopefully :smiley: )